x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawl_select_domains_modal/crawl_select_domains_modal.tsx

```diff
@@ -0,0 +1,107 @@
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiNotificationBadge,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { CANCEL_BUTTON_LABEL } from '../../../../../shared/constants';
+
+import { CrawlerLogic } from '../../crawler_logic';
+
+import { CrawlSelectDomainsModalLogic } from './crawl_select_domains_modal_logic';
+import { SimplifiedSelectable } from './simplified_selectable';
+
+import './crawl_select_domains_modal.scss';
+
+export const CrawlSelectDomainsModal: React.FC = () => {
+  const { domains } = useValues(CrawlerLogic);
+  const domainUrls = domains.map((domain) => domain.url);
+
+  const crawlSelectDomainsModalLogic = CrawlSelectDomainsModalLogic({ domains });
+  const { isModalVisible, selectedDomainUrls } = useValues(crawlSelectDomainsModalLogic);
+  const { hideModal, onSelectDomainUrls } = useActions(crawlSelectDomainsModalLogic);
+
+  const { startCrawl } = useActions(CrawlerLogic);
+
+  if (!isModalVisible) {
+    return null;
+  }
+
+  return (
+    <EuiModal onClose={hideModal} className="crawlSelectDomainsModal">
+      <EuiModalHeader>
+        <EuiFlexGroup alignItems="center" gutterSize="m">
+          <EuiFlexItem>
+            <EuiModalHeaderTitle>
+              {i18n.translate(
+                'xpack.enterpriseSearch.appSearch.crawler.crawlSelectDomainsModal.modalHeaderTitle',
+                {
+                  defaultMessage: 'Crawl select domains',
+                }
+              )}
+            </EuiModalHeaderTitle>
+          </EuiFlexItem>
+          <EuiNotificationBadge
+            size="m"
+            color={selectedDomainUrls.length > 0 ? 'accent' : 'subdued'}
+          >
+            {selectedDomainUrls.length}
+          </EuiNotificationBadge>
+          <EuiFlexItem grow={false}>
+            {i18n.translate(
+              'xpack.enterpriseSearch.appSearch.crawler.crawlSelectDomainsModal.selectedDescriptor',
+              {
+                defaultMessage: 'selected',
+              }
+            )}
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <SimplifiedSelectable
+          options={domainUrls}
+          selectedOptions={selectedDomainUrls}
+          onChange={onSelectDomainUrls}
+        />
+      </EuiModalBody>
+      <EuiModalFooter>
+        <EuiButtonEmpty onClick={hideModal}>{CANCEL_BUTTON_LABEL}</EuiButtonEmpty>
+        <EuiButton
+          fill
+          onClick={() => {
+            startCrawl({ domain_allowlist: selectedDomainUrls });
```

Comment:
question: Do we generally immediately close modals or move on for these actions, or do we wait for API calls to succeed?

Reply:
_It depends_, with two wrinkles to consider:

1) Is the logic to make the API call in the same logic as the closeModal function? If yes, pretty straight forward.  If no, you have to decide how you're going to couple these.  `startCrawl` is a part of `CrawlerLogic` which represents the top-level logic for this section of the app.  I think the `GenericEndpointInline...` components represent how confusing it can get trying to pass around actions for different logic files around as "callbacks".  Not clear to me what a good pattern was `specific component -> calls action in top level logic file -> triggers action in specific component` would be.  
2) Our error messaging is mostly done in Flash Messages at the page level, rather than putting feedback directly in the specific modal/flyout/etc.  So regardless of success or failure, the messaging is going to be on the page, so we need to close the modal/flyout/etc anyway.  It's kind of a similar problem to #1 above, how do we handle an `onError` callback style action in kea if the action is in a more generic logic file?

It's much easier to do this if the API calls and UX actions are in the same file. Would it make sense to duplicate the `startCrawl` API call in each component that needs to start a crawl? I'm also not sure! 

I think the ideal situation would be:
1) user clicks save
2) save button disables/replaced with spinner, modal stays open while waiting for response
3) on error, modal stays open and reports error inside modal (+++++ highlights error fields)
4) on success, close modal, provide success feedback
but its a corner we've often cut.

Reply:
There's a pattern I've used in the past with other Redux implementations that allows you to do this fairly easily without callbacks and without having everything in the same file, which involves firing a success or error action when you do the API call. So in addition to `startCrawl` you'd have a `startCrawlSuccess` and `startCrawlError` action in your `CrawlerLogic` that get fired depending on the result of the call. That would allow you to listen for `startCrawlSuccess` to fire `hideModal` and `startCrawlError` to show an error (and stop the spinner).

Reply:
Would love to see this in action!